### PR TITLE
Refactor profile image

### PR DESCRIPTION
**Pull request by tdunbar123**
Validate file extensions on upload image

### CODE REVIEW

**Comment by escalonn** [comment]
request review from kenneth anytime you do from me

**Review by escalonn — CHANGES_REQUESTED** [review]
the failing tests should be added first in their own separate PR, not to main branch, but to the branch that will be itself merged in a PR to main later.
then the implementation in another separate PR.

fyi this could be better - we should be thinking about JPG and jpg and JPEG etc, including tests with Theory for those cases. once those tests exist and fail, then code can be written (in another PR) accordingly

**Review by escalonn — CHANGES_REQUESTED** [review]

File: Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfilePictureController.cs [hunk 1]
```diff
@@ -57,10 +58,32 @@ public async Task<ActionResult> Post()
...
+                "SVG", 
+                "WEBP", 
+                "AVIF", 
+                "APNG"
+            };
+            // validate file extension to be valid image
+            if (!validExtensions.Contains(extension))
+            {
+                _logger.LogError("File is not a valid image.");
+                return BadRequest();
```
[the remark this answers is not in the document]
Comment (modem428) [reply Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfilePictureController.cs]: the log and error message should have the extension included

[the remark this answers is not in the document]
Comment (tdunbar123) [reply Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfilePictureController.cs]: Included bad extension in logging info and BadRequest message.

**Review by escalonn — APPROVED** [review]

**Pull request merged**